src/tactic/squeeze.lean

```diff
@@ -148,8 +160,18 @@ do some s ← get_proof_state_after (tac ff (user_args ++ simp_args)),
    pure (user_args' ++ simp_args')
 
 /-- make a `simp_arg_type` that references the name given as an argument -/
+-- this is currently very badly written - using an `expr` constructor raw is not a good idea
 meta def name.to_simp_args (n : name) : tactic simp_arg_type :=
-do e ← resolve_name' n, pure $ simp_arg_type.expr e
+return $ simp_arg_type.expr $ @expr.local_const ff n n (default) (expr.var 0)
+
+-- `macro`s can be other things but this is a good first order approximation
```

Comment:
You could add this comment to the doc string and explain what the first-order approximation is.